### PR TITLE
Varvara/feature/client registration

### PR DESCRIPTION
**Pull request by varvarashcherbyna7**

### CODE REVIEW

File: src/api/authApi.ts [hunk 1]
```diff
@@ -42,49 +20,74 @@ const formatRequestBodyApiKey = (password: string, api_key: string) => {
   return params;
 };
 
+interface IPatientFormForBackend {
+  firstName: string,
+  lastName: string,
+  birthday: string,
```
Comment (kipod) [inline src/api/authApi.ts:26]:
```typescrypt
birthday: Date,
```

File: src/api/authApi.ts [hunk 2]
```diff
@@ -42,49 +20,74 @@ const formatRequestBodyApiKey = (password: string, api_key: string) => {
   return params;
 };
 
+interface IPatientFormForBackend {
```
Comment (kipod) [inline src/api/authApi.ts:23]: IMHO this is class, and authApi.ts is not a good place for it...

File: src/api/authApi.ts [hunk 3]
```diff
@@ -42,49 +20,74 @@ const formatRequestBodyApiKey = (password: string, api_key: string) => {
...
-      })
-      .catch((error: AxiosError<ILoginResponse>) => {
-        // place to handle errors and rise custom errors
-        console.log(`POST [api/auth/sign_in] error message: ${error.message}`);
-        throw error.message;
-      });
-    return response;
+       console.log("POST [/auth/sign_in] response received successfully");
+      return response.data;
+    } catch (error: any) {
```
Comment (kipod) [inline src/api/authApi.ts:78]: catch all?? can you have here define error type??

Comment (kipod) [reply src/api/authApi.ts:78]: ```} catch (error) {```

File: src/api/authApi.ts [hunk 4]
```diff
@@ -1,90 +1,118 @@
...
-        // place to handle errors and rise custom errors
-        console.log(`POST [api/sing_up/${api_key}] error message: ${error.message}`);
-        throw error.message;
-      });
-    return response;
+      return response.data;
+    } catch (error: any) {
+      // place to handle errors and rise custom errors
+      console.log(`POST [api/sing_up/${api_key}] error message: ${error.message}`);
+      throw error.message;
```
Comment (kipod) [inline src/api/authApi.ts:94]: ```throw new Error(error.message);```

File: src/types/patientsTypes.ts [hunk 5]
```diff
@@ -3,4 +3,38 @@ export interface IPatient {
   name: string;
   lastName: string;
   phone: string;
+};
+
+export interface IPatientForm {
```
Comment (kipod) [inline src/types/patientsTypes.ts:8]: Stop, I sow that in interface IPatientFormForBackend

Is it duplication?

**Review by kipod — COMMENTED** [review]
OK. Thnak you

File: src/api/authApi.ts (hunk 1 above)
Comment (varvarashcherbyna7) [reply src/api/authApi.ts:26]: here it is necessary to specify term as on beck the pydantic from term automatically converts to date

File: src/types/patientsTypes.ts (hunk 5 above)
Comment (varvarashcherbyna7) [reply src/types/patientsTypes.ts:8]: I delete it )

Comment (varvarashcherbyna7) [reply src/types/patientsTypes.ts:8]: it is not a duplicate
this is to change the types by some value

**Pull request merged**